### DISCUSSION TITLE
API method to get all alerts

### DISCUSSION THREAD

**Discussion by greatvovan** [grafana/grafana#55601] [Alerting]
Looking at [Alerting provisioning API](https://grafana.com/docs/grafana/latest/developers/http_api/alerting_provisioning/), I cannot figure out how to get list of all alerts (or at least list of their IDs). A REST API is unimaginable without `GET /items` method, but I cannot find it on the doc page. Just blindly trying to `GET /api/v1/provisioning/alert-rules` does not work either (returning 404).

How can I obtain the alert list or at least reset all alerts to be able to re-create them from scratch?

**Comment by greatvovan** [comment]
Support helped me find [this API](https://editor.swagger.io/?url=https://raw.githubusercontent.com/grafana/grafana/main/pkg/services/ngalert/api/tooling/post.json) where `/api/ruler/grafana/api/v1/rules` gives rule UIDs among other things, however it is a different API. Should not provisioning API have this method? Contact points, notification policies, mute timings, and templated do have “GET all” methods. Why don’t alert rules have one?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!

**Reply by greatvovan** [reply]
FYI, this was not a feature request.

**Reply by gelicia** [reply]
@greatvovan Like I said in the other ticket, my apologies. Please feel free to make this an issue and tag it `area/alerting` and `area/backend/api`
